src/pages/Register.jsx

```diff
@@ -52,32 +52,39 @@ function Register () {
     }, [email])
 
     useEffect(() => {
-        const isPwdValid = PWD_REGEX.test(pwd);
+        const isPwdValid = PWD_REGEX.test(password);
         setValidPwd(isPwdValid);
-        setValidMatch(isPwdValid && pwd === matchPwd);
-    }, [pwd, matchPwd])
+        setValidMatch(isPwdValid && password === matchPwd);
+    }, [password, matchPwd])
 
     useEffect(() => {
         setErrMsg('');
-    }, [firstName, lastName, email, pwd, matchPwd])
-
+    }, [firstName, lastName, email, password, matchPwd])
+    
     const handleSubmit = async (e) => {
         e.preventDefault();
         
-        try {  
+        try { 
+            // Send registration data to the backend 
             const response = await axiosInstance.post('/auth/register', 
-                JSON({firstName, lastName, email, pwd }),
+                JSON.stringify({firstName, lastName, email, password }),
                 {
                     headers: {'Content-Type': 'application/json'}
                 }
             );
+            
+            // Extract the token from the response
+            const token = response.data.token;        
```

Comment:
I don't think the token comes back when you register, only when you login successfully. Currently this is returning an undefined object (I checked the local storage after trying to run this).

You'll need to do the jwt storage in the `LogIn` component instead after successful authentication